src/integrators/sdirk_integrators.jl

```diff
@@ -518,67 +556,75 @@ end
     W = 1 - dto2*J
   end
 
+  # initial guess
+  zprev = dt.*integrator.fsalfirst
   z = zprev # Constant extrapolation
 
-  iter = 0
-  κ = cache.κ
-  tol = cache.tol
-
-  iter += 1
-  u = uprev + (zprev + z)/2
-  b = -z .+ dt.*f(t+dt,u)
+  # initial step of Newton iteration
+  iter = 1
+  tstep = t + dt
+  tmp = @. uprev + dto2*integrator.fsalfirst
+  u = @. tmp + 1//2*z
```

Comment:
`/2` instead, this doesn't compile as well as I'd hope. At least it used to do some bizzare things.